src/site-switcher/siteSwitcherComponent.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Controller for site switcher component
+ * @memberof miqStaticAssets
+ * @ngdoc controller
+ * @name SiteSwitcherController
+ */
+export class SiteSwitcherController {
```

Comment:
Personally, I'm not wild about the `any` in a public interface. I don't think private methods need any typing, but public interfaces are where we are trying to control the interchange of data amongst various clients (internal and external). For instance, we shouldn't allow `1` to be passed to iconClass. Ideally, it would be an `enum` of our iconClasses that would instantly pop-up as we type the iconClass so that invalid icons would be caught at compile time not runtime. 

The enums don't need to be in this PR, but just thoughts for later. Unless people don't agree with these enums typing at all? Which we should discuss now.